doc/content/system/scheduler.md

```diff
@@ -1,4 +1,68 @@
 # Scheduler {#system-schedulers}
 
-\warning
-The scheduler system is pending documentation.
+[TOC]
+
+## Heterogeneous computing
+
+Modern computing architectures are typically heterogenous: the computing power on a node is divided across multiple asymmetric computing devices.  A typical example is a cluster where each node includes one or more CPUs, providing compute with multiple threads that all share memory, and one or more GPUs, which may or may not share memory with each other and the CPUs.  NEML2 provides software infrastructure to efficiently divide, send, recover, and merge batched constitutive model updates across multiple asymmetric compute devices.
+
+\note
+Currently NEML2 only supports CPU and CUDA compute devices.  Additional accelerators supported by pytorch including Apple [mps](https://pytorch.org/docs/stable/mps.html) and Intel [xpu](https://pytorch.org/docs/stable/xpu.html) may also work, but are not officially supported.
+
+NEML2 manages work scheduling, dispatch, and joining with two types of objects: dispatchers and schedulers.  These objects divide up work along a single batch axis by subdividing that axis into smaller chunks (subbatches) to run on a given device.
+
+## Work dispatchers
+
+NEML2 currently only has a single type of work dispatcher.  The dispatcher maintains a main thread which schedules and distributes work according the algorithm provided by the selected work scheduler.  The main thread dispatches the work to a thread pool that maintains a specific thread for each device.  The thread picks up the task (representing a batch of work), sends the work to the device, and returns the work, once completed, to the main thread.
+
+Setting up the work dispatcher requires providing it with lambda functions describing how to:
+1. Actually run a subbatch of work on a device
+2. Reduce the final collection of subbatches into a single result
+3. Preprocess a subbatch before running it (for example, send it to the device)
+4. Postprocess a subbatch before joining (for example, send it back to the cpu)
+5. Initialize the device thread.
```

Comment:
Seems like it should be sub-batch:
https://books.google.com/ngrams/graph?content=subbatch%2Csub-batch&year_start=1940&year_end=2022&corpus=en&smoothing=3&case_insensitive=false